### PR TITLE
Added 5th Track + Bug Fix.

### PR DESCRIPTION
**Pull request by vasil7112**

### CODE REVIEW

**Comment by ONSBalder** [comment]
Can one of the admins verify this patch?

**Comment by vasil7112** [comment]
Wow ONSBalder that was instant :smile:

File: obs/adv-audio-control.cpp [hunk 1]
```diff
@@ -35,6 +35,7 @@ OBSAdvAudioCtrl::OBSAdvAudioCtrl(QGridLayout *layout, obs_source_t *source_)
 	mixer2                         = new QCheckBox();
 	mixer3                         = new QCheckBox();
 	mixer4                         = new QCheckBox();
+	mixer5						   = new QCheckBox();
```
Comment (robske110) [inline obs/adv-audio-control.cpp]: formatting?

File: obs/adv-audio-control.cpp [hunk 2]
```diff
@@ -120,6 +124,8 @@ OBSAdvAudioCtrl::OBSAdvAudioCtrl(QGridLayout *layout, obs_source_t *source_)
 			this, SLOT(mixer3Changed(bool)));
 	QWidget::connect(mixer4, SIGNAL(clicked(bool)),
 			this, SLOT(mixer4Changed(bool)));
+	QWidget::connect(mixer5, SIGNAL(clicked(bool)),
+		this, SLOT(mixer5Changed(bool)));
```
Comment (robske110) [inline obs/adv-audio-control.cpp]: formatting again

File: obs/forms/OBSBasicSettings.ui [hunk 3]
```diff
@@ -2333,6 +2354,131 @@
                       </layout>
                      </widget>
                     </item>
+
+
```
Comment (robske110) [inline obs/forms/OBSBasicSettings.ui]: too much newlines?

File: obs/window-basic-settings.cpp [hunk 4]
```diff
@@ -301,6 +302,7 @@ OBSBasicSettings::OBSBasicSettings(QWidget *parent)
 	HookWidget(ui->advOutTrack2,         CHECK_CHANGED,  OUTPUTS_CHANGED);
 	HookWidget(ui->advOutTrack3,         CHECK_CHANGED,  OUTPUTS_CHANGED);
 	HookWidget(ui->advOutTrack4,         CHECK_CHANGED,  OUTPUTS_CHANGED);
+	HookWidget(ui->advOutTrack5,		 CHECK_CHANGED,  OUTPUTS_CHANGED);
```
Comment (robske110) [inline obs/window-basic-settings.cpp]: formatting

File: obs/window-basic-settings.cpp [hunk 5]
```diff
@@ -314,6 +316,7 @@ OBSBasicSettings::OBSBasicSettings(QWidget *parent)
 	HookWidget(ui->advOutRecTrack2,      CHECK_CHANGED,  OUTPUTS_CHANGED);
 	HookWidget(ui->advOutRecTrack3,      CHECK_CHANGED,  OUTPUTS_CHANGED);
 	HookWidget(ui->advOutRecTrack4,      CHECK_CHANGED,  OUTPUTS_CHANGED);
+	HookWidget(ui->advOutRecTrack5,		 CHECK_CHANGED,  OUTPUTS_CHANGED);
```
Comment (robske110) [inline obs/window-basic-settings.cpp]: formatting

File: obs/window-basic-settings.cpp [hunk 6]
```diff
@@ -330,6 +333,7 @@ OBSBasicSettings::OBSBasicSettings(QWidget *parent)
 	HookWidget(ui->advOutFFTrack2,       CHECK_CHANGED,  OUTPUTS_CHANGED);
 	HookWidget(ui->advOutFFTrack3,       CHECK_CHANGED,  OUTPUTS_CHANGED);
 	HookWidget(ui->advOutFFTrack4,       CHECK_CHANGED,  OUTPUTS_CHANGED);
+	HookWidget(ui->advOutFFTrack5,		 CHECK_CHANGED,  OUTPUTS_CHANGED);
```
Comment (robske110) [inline obs/window-basic-settings.cpp]: formatting

File: obs/window-basic-settings.cpp [hunk 7]
```diff
@@ -340,6 +344,8 @@ OBSBasicSettings::OBSBasicSettings(QWidget *parent)
 	HookWidget(ui->advOutTrack3Name,     EDIT_CHANGED,   OUTPUTS_CHANGED);
 	HookWidget(ui->advOutTrack4Bitrate,  COMBO_CHANGED,  OUTPUTS_CHANGED);
 	HookWidget(ui->advOutTrack4Name,     EDIT_CHANGED,   OUTPUTS_CHANGED);
+	HookWidget(ui->advOutTrack5Bitrate,  COMBO_CHANGED,  OUTPUTS_CHANGED);
+	HookWidget(ui->advOutTrack5Name,	 EDIT_CHANGED,	 OUTPUTS_CHANGED);
```
Comment (robske110) [inline obs/window-basic-settings.cpp]: formatting

File: obs/window-basic-settings.cpp [hunk 8]
```diff
@@ -2464,7 +2488,8 @@ void OBSBasicSettings::SaveOutputSettings()
 	SaveCombo(ui->advOutRescale, "AdvOut", "RescaleRes");
 	SaveTrackIndex(main->Config(), "AdvOut", "TrackIndex",
 			ui->advOutTrack1, ui->advOutTrack2,
-			ui->advOutTrack3, ui->advOutTrack4);
+			ui->advOutTrack3, ui->advOutTrack4,
+		    ui->advOutTrack5);
```
Comment (robske110) [inline obs/window-basic-settings.cpp]: formatting

File: obs/window-basic-settings.cpp [hunk 9]
```diff
@@ -2502,16 +2528,19 @@ void OBSBasicSettings::SaveOutputSettings()
 	SaveEdit(ui->advOutFFACfg, "AdvOut", "FFACustom");
 	SaveTrackIndex(main->Config(), "AdvOut", "FFAudioTrack",
 			ui->advOutFFTrack1, ui->advOutFFTrack2,
-			ui->advOutFFTrack3, ui->advOutFFTrack4);
+			ui->advOutFFTrack3, ui->advOutFFTrack4,
+		    ui->advOutFFTrack5);
```
Comment (robske110) [inline obs/window-basic-settings.cpp]: formatting

File: obs/window-basic-settings.cpp [hunk 10]
```diff
@@ -3159,6 +3189,7 @@ void OBSBasicSettings::UpdateAdvOutStreamDelayEstimate()
 	case 2: aBitrateText = ui->advOutTrack2Bitrate->currentText(); break;
 	case 3: aBitrateText = ui->advOutTrack3Bitrate->currentText(); break;
 	case 4: aBitrateText = ui->advOutTrack4Bitrate->currentText(); break;
+    case 5: aBitrateText = ui->advOutTrack5Bitrate->currentText(); break;
```
Comment (robske110) [inline obs/window-basic-settings.cpp]: formatting

**Comment by robske110** [comment]
I think guthub messed up your formatting, please check that :)

**Comment by Nightfirecat** [comment]
Doesn't look like a GitHub problem; I think your editor is configured to translate 4 spaces into a tab.

**Comment by vasil7112** [comment]
Indeed it is not Github, its most likely my settings.

**Comment by robske110** [comment]
Well, try to fix it anyways

**Comment by vasil7112** [comment]
I mean that it should probably be different for each user and not the same. For example OBS for me was formatted differently than other projects. So i guess after all its just what each one enjoys.

Thanks though for trying to make it "compatible" for the master OBS ;)

**Comment by robske110** [comment]
Well, I know the Formatting problem a lot too. I always locally code in Tabs and size 4 github is space + size 8
And sometimes it just doesn't work, allthough it is displayed correctly for me. If that happens i just put my code under a formatter, wich i usally set to tabs. Now some spaces that i couldn't see due to they looked exactly like the tabs are transformed into tabs don't mess up github. Annyoing, but i soon found that my beloved text editor had 'Indent selection'.

**Comment by ElectronicWar** [comment]
Well Studio has pretty strict coding guidelines to avoid the mess that Classic was code-wise.
Please read up here: https://github.com/jp9000/obs-studio/blob/master/CONTRIBUTING

**Comment by vasil7112** [comment]
Thanks for the info, but this specific Pull Request will not make it to the OBS. I talked with the Jim (The Developer) and he doesn't like the idea of more tracks than how many there actually are.

Again though, thanks for the info :)

**Comment by jp9000** [comment]
Yea, although I understand that there are some who may specifically have a case for 5 tracks, I personally wish to avoid adding more tracks for now.  (And although some might be quick to say "just make it variable", the implications of such from both a performance, maintenance, and user interface perspective aren't something that I really feel are worth dealing with -- at this point in time at least)

**Pull request closed without merging**